azurelinuxagent/common/osutil/default.py

```diff
@@ -267,11 +276,22 @@ def enable_firewall(self, dst_ip, uid):
 
         try:
             wait = self.get_firewall_will_wait()
-
             # If the DROP rule exists, make no changes
             try:
                 drop_rule = _get_firewall_drop_command(wait, "-C", dst_ip)
                 shellutil.run_command(drop_rule)
+                try:
+                    accept_rule = _get_firewall_accept_command_nonroot_tcp(wait, "-C", dst_ip)
+                    shellutil.run_command(accept_rule)
+                except CommandError as e:
+                    logger.info("Iptable rule to allow tcp request to wireserver is not found. Adding it")
+                    try:
+                        accept_rule = _get_firewall_accept_command_nonroot_tcp(wait, "-I", dst_ip)
```

Comment:
This part looks very clunky, could you modularize it to make it a bit cleaner? Thanks!

Reply:
Also, I see you added a new parameter `__INSERT_COMMAND` in networkutil.py file, can you use that here too please (might need to make that constant public to use it in this module)

Reply:
After having a discussion with Norberto , this logic moved to Update.py